go.mod

```diff
@@ -26,7 +26,7 @@
 
 module github.com/openthread/ot-ns
 
-go 1.18
+go 1.23
```

Comment:
This specifies the minimum supported Go version. If we'd like to increase that, then also related documentation needs to be updated.
If we go to higher versions, then keeping the minimum required version at 1.18 may still be useful e.g. for older distros/OSes.

Reply:
I was exploring what it would take to fix the failures in https://github.com/openthread/ot-ns/pull/541

Dependabot created this PR to address reported security vulnerabilities in the underlying dependencies.